### ISSUE TITLE
Consider using -# Catch2 flag for tagging test cases

### ISSUE THREAD

**Issue by horenmar** [microsoft/vcpkg#11006]
Right now, the vcpkg-maintainer documentation tells people that they need to tag `TEST_CASE`s with tag of `[filename-but-without-the-cpp-suffix]` tag. This is then used to register tests from each TU separately with CTest.

Catch2 provides `-#` flag, which automatically adds `[#filename-but-without-last-extension]` as a tag to each `TEST_CASE`, which could be used to automate the process above, and decrease reviewer/maintainer burden.

**Comment by horenmar** [comment]
Okay, after some extra investigations, the tag is currently unused (but AFAIK it was used in the past), and instead all tests are run in single invocation. This means that the tags are mostly pointless, and also that the test runs for ~8s (on my machine) without any option to run them in parallel.

🤷

**Comment by JackBoosY** [comment]
Duplicated to #11193.

**Comment by horenmar** [comment]
What? No, this is in no way a duplicate of 11193. #11193 is about running unit tests of the _ports_ (the packages vcpkg provides). This issue is about running the _vcpkg's own_ unit test in a smarter manner.

**Comment by horenmar** [comment]
I noticed that the actual vcpkg sources have moved to a different repository, so keeping this issue here is pointless.

**Comment by strega-nil** [comment]
@horenmar sorry I didn't see this, but this is where issues go for vcpkg-tool; could you explain how to use -#? We actually do allow people to run things on their own, we just don't _really_ support ctest very well. Afaik, we (vcpkg devs) all use vcpkg-test manually.

**Comment by horenmar** [comment]
@strega-nil So, this is mostly context from when I opened the issue (I am too lazy to go digging through the docs and sources again 😄 ):

The developer docs for vcpkg-tool used to say that every `TEST_CASE` in file `foo.cpp` should be tagged with `[foo]`. Catch2 can do that automatically for you, if you run the binary with `-#` flag: running `./vcpkg-tests -# --list-tags` will show that every `TEST_CASE` is now tagged with `[#filename-without-last-extension]` tag.

You can then also use this to simply generate test batches for CTest and for running them in parallel -- last I checked the tests were already taking multiple seconds to run in series.
